src/libraries/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs

```diff
@@ -3103,4 +3103,95 @@ private static string GetAmString(DateTimeFormat dateTimeFormat)
     {
         return ((CultureInfo)dateTimeFormat.FormatProvider).DateTimeFormat.AMDesignator;
     }
+
+    // Test for the fix to ensure DataContractJsonSerializer passes ISerializationSurrogateProvider to internal XML serializer
+    [Fact]
+    public static void DCJS_SerializationSurrogateProvider_PassedToInternalSerializer()
+    {
+        // Setup: Create a test surrogate that transforms TestClassA to TestClassB
+        var surrogateProvider = new TestSurrogateProvider();
+        
+        // Create the serializer and set the surrogate provider
+        var serializer = new DataContractJsonSerializer(typeof(TestClassA));
+        serializer.SetSerializationSurrogateProvider(surrogateProvider);
+        
+        var testObj = new TestClassA { Value = "OriginalValue" };
+        
+        // Act: Serialize the object
+        byte[] serializedData;
+        using (var ms = new MemoryStream())
+        {
+            serializer.WriteObject(ms, testObj);
+            serializedData = ms.ToArray();
+        }
+        
+        // The surrogate should have been called during serialization
+        Assert.True(surrogateProvider.SerializationSurrogateWasCalled, "Serialization surrogate should have been called during WriteObject");
+        
+        // Act: Deserialize the object
+        TestClassA deserializedObj;
+        using (var ms = new MemoryStream(serializedData))
+        {
+            deserializedObj = (TestClassA)serializer.ReadObject(ms);
+        }
+        
+        // The surrogate should have been called during deserialization
+        Assert.True(surrogateProvider.DeserializationSurrogateWasCalled, "Deserialization surrogate should have been called during ReadObject");
+        
+        // Verify the surrogate transformation was applied
+        Assert.Equal("TransformedValue", deserializedObj.Value);
+    }
+
+    // Test classes for surrogate provider testing
+    [DataContract]
+    public class TestClassA
+    {
+        [DataMember]
+        public string Value { get; set; }
+    }
+
+    [DataContract]
+    public class TestClassB
+    {
+        [DataMember]
+        public string TransformedValue { get; set; }
+    }
+
+    // Test surrogate provider implementation
+    public class TestSurrogateProvider : ISerializationSurrogateProvider
```

Comment:
Instead of introducing new test classes, try to reuse the `NonSerializablePerson`, `NonSerializablePersonSurrogate`, and `MyPersonSurrogateProvider` from the shared file '' in place of `TestClassA`, `TestClassB`, and `TestSurrogateProvider`. You are allowed to add the '*WasCalled' properties to `MyPersonSurrogateProvider`.